devito/core/gpu_openmp.py

```diff
@@ -321,6 +400,9 @@ def _specialize_iet(cls, graph, **kwargs):
         data_manager.place_definitions(graph)
         data_manager.place_casts(graph)
 
+        # Initialize OpenMP environment
+        initialize(graph)
```

Comment:
I think this should be the first pass of the pipeline, before the "Distributed-memory parallelism" block

Reply:
Doing that generates incorrect C code. Also, that is the same position `initialize` is in `gpu_openacc.py`.